cmd/resourcesnapshot/resourcesnapshot_restore.go

```diff
@@ -0,0 +1,108 @@
+package resourcesnapshot
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/civo/civogo"
+	"github.com/civo/cli/common"
+	"github.com/civo/cli/config"
+	"github.com/civo/cli/utility"
+	"github.com/spf13/cobra"
+)
+
+var (
+	restoreDescription string
+	restoreHostname    string
+	restorePrivateIPv4 string
+	includeVolumes     bool
+	overwriteExisting  bool
+)
+
+var resourceSnapshotRestoreCmd = &cobra.Command{
+	Use:     "restore ID/NAME",
+	Args:    cobra.MinimumNArgs(1),
+	Aliases: []string{"recover"},
+	Short:   "Restore from a resource snapshot",
+	Long: `Restore a resource from a snapshot by ID or name.
+If you wish to use a custom format, the available fields are:
+
+* id
+* name
+* description
+* resource_type`,
+	Run: func(cmd *cobra.Command, args []string) {
+		utility.EnsureCurrentRegion()
+
+		client, err := config.CivoAPIClient()
+		if common.RegionSet != "" {
+			client.Region = common.RegionSet
+		}
+		if err != nil {
+			utility.Error("Creating the connection to Civo's API failed with %s", err)
+			os.Exit(1)
+		}
+
+		// Get the snapshot first to verify it exists
+		snapshot, err := client.GetResourceSnapshot(args[0])
```

Comment:
there should be the check on len(args) or it could panic

Reply:
https://github.com/civo/cli/pull/533/commits/10f6e7a1f155f335510be3dc1a150bb1b7278034

Reply:
sorry, can you rollback? I realized it is already ensured in here: `cobra.MinimumNArgs(1)`

Reply:
(just drop the latest commit)

Reply:
Done